plugin/src/main/java/org/opensearch/ml/rest/RestMLPredictionAction.java

```diff
@@ -103,7 +103,7 @@ public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client
                 client
                     .execute(
                         MLPredictionTaskAction.INSTANCE,
-                        getRequest(modelId, algoName, request),
+                        getRequest(modelId, functionName.get().name(), algoName, request),
```

Comment:
This can break here. Because what if model isn't in cache. And it's just trying to pull model information from the system index?

Reply:
If the model isn't deployed yet then we will find function name as null, and then algoName as "Remote" and this won't work.